### PR TITLE
Abstract __add__ for Profilers

### PR DESCRIPTION
**Pull request by JGSweets**
Merges similarities between the __add__ functions for both Structured and Unstructured and creates the abstract form in the BaseProfiler.

### CODE REVIEW

**Review by ChrisWallace2020 — DISMISSED** [review]

File: dataprofiler/profilers/profile_builder.py [hunk 1]
```diff
@@ -502,15 +502,40 @@ def __init__(self, data, samples_per_update=None, min_true_samples=None,
...
+
+        merged_profile = self.__class__(
+            data=None, samples_per_update=self._samples_per_update,
+            min_true_samples=self._min_true_samples, options=self.options
+        )
+        merged_profile.encoding = self.encoding \
+            if self.encoding == other.encoding else 'multiple files'
+        merged_profile.file_type = self.file_type \
+            if self.file_type == other.file_type else 'multiple files'
+        merged_profile.total_samples = self.total_samples + other.total_samples
```
Comment (lettergram) [inline dataprofiler/profilers/profile_builder.py:536]: can we break these into normal if / else looking statements. It's quite difficult to read these multi-line if / else.

Comment (JGSweets) [reply dataprofiler/profilers/profile_builder.py:536]: as in this?
```python
merged_profile.encoding = self.encoding
if self.encoding != other.encoding:
    self.encoding = 'multiple files'

merged_profile.file_type = self.file_type
if self.file_type != other.file_type:
    self.file_type = 'multiple files'
```

File: dataprofiler/profilers/profile_builder.py [hunk 2]
```diff
@@ -1103,41 +1132,40 @@ def __init__(self, data, samples_per_update=None, min_true_samples=0,
...
+        """
+        Merges two profiles together overriding the `+` operator.
+
+        :param other: profile being added to this one.
+        :type other: StructuredProfiler
+        :return: merger of the two profiles
+        :rtype: StructuredProfiler
+        """
+        merged_profile = super().__add__(other
+                                         )
```
Comment (AnhTruong) [inline dataprofiler/profilers/profile_builder.py:1159]: can we remove ) to the above line

**Review by AnhTruong — DISMISSED** [review]

File: dataprofiler/profilers/profile_builder.py (hunk 2 above)
Comment (JGSweets) [reply dataprofiler/profilers/profile_builder.py]: fixed, good catch

**Review by ChrisWallace2020 — DISMISSED** [review]

**Review by AnhTruong — DISMISSED** [review]

**Review by ChrisWallace2020 — APPROVED** [review]

**Review by AnhTruong — APPROVED** [review]

**Pull request merged**